cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/remote/RemoteExecutionTwillRunnerService.java

```diff
@@ -215,10 +224,23 @@ public TwillPreparer prepare(TwillApplication application) {
     RunId twillRunId = RunIds.generate();
     String remoteHost = masterNode.getProperties().get("ip.external");
 
+    Optional<ProfileId> profile = SystemArguments.getProfileIdFromArgs(programRunId.getNamespaceId(),
+                                                                         programOptions.getArguments().asMap());
+    ProfileId profileId =
+      profile.orElseThrow(() -> new IllegalStateException(String.format("Missing profile information for this " +
```

Comment:
Just use `+`. No need to use string format as what you are appending is at the end.

Reply:
Also, suggest to break the line like this:

```java
ProfileId profileId = profile.orElseThrow(() -> 
  new IllegalStateException("Missing profile information for program run " + programRunId));
```